src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -798,201 +915,183 @@ def get_traces(self, start_frame, end_frame, channel_indices):
-                        f"Available options are: {available_electrical_series}"
-                    )
-            self.electrical_series_location = available_electrical_series[electrical_series_name]
-            electrical_series = hdf5_file[self.electrical_series_location]
+        When a `t_start` is not provided it will be inferred from the corresponding ElectricalSeries with name equal
+        to `electrical_series_path`. The `t_start` then will be either the `ElectricalSeries.starting_time` or the
+        first timestamp in the `ElectricalSeries.timestamps`.
+    storage_options: dict | None = None,
+        Additional parameters for the storage backend (e.g. AWS credentials) used for "zarr" stream_mode.
+    use_pynwb: bool, default: False
+        Uses the pynwb library to read the NWB file. Setting this to False, the default, uses h5py
+        to read the file. Using h5py can improve performance by bypassing some of the PyNWB validations.
 
-            # Get sampling frequency
-            if "starting_time" in electrical_series.keys():
-                t_start = electrical_series["starting_time"][()]
-                sampling_frequency = electrical_series["starting_time"].attrs["rate"]
-            elif "timestamps" in electrical_series.keys():
-                timestamps = electrical_series["timestamps"][:]
-                t_start = timestamps[0]
-                sampling_frequency = 1.0 / np.median(np.diff(timestamps[:samples_for_rate_estimation]))
+    Returns
+    -------
+    sorting: NwbSortingExtractor
+        The sorting extractor for the NWB file.
+    """
 
-            self.t_start = t_start
+    extractor_name = "NwbSorting"
+    mode = "file"
+    installation_mesg = "To use the Nwb extractors, install pynwb: \n\n pip install pynwb\n\n"
+    name = "nwb"
 
-        assert (
-            sampling_frequency is not None
-        ), "Couldn't load sampling frequency. Please provide it with the 'sampling_frequency' argument"
-        assert (
-            self.t_start is not None
-        ), "Couldn't load a starting time for the sorting. Please provide it with the 't_start' argument"
+    def __init__(
+        self,
+        file_path: str | Path,
+        electrical_series_path: str | None = None,
+        sampling_frequency: float | None = None,
+        samples_for_rate_estimation: int = 1_000,
+        unit_table_path: str = "units",
+        stream_mode: str | None = None,
+        stream_cache_path: str | Path | None = None,
+        *,
+        t_start: float | None = None,
+        cache: bool = False,
+        storage_options: dict | None = None,
+        use_pynwb: bool = False,
+    ):
+        self.stream_mode = stream_mode
+        self.stream_cache_path = stream_cache_path
+        self.electrical_series_path = electrical_series_path
+        self.file_path = file_path
+        self.t_start = t_start
+        self.internal_sampling_frequency = sampling_frequency
```

Comment:
why internal sampling frequency? in opposition to external?

Reply:
initially it was `self._sampling_frequency` (or `self.sampling_frequency`), but these are used by the `BaseSorting` object, so we need another name.

Reply:
Ah, got you. It is for having it as an attribute. What about `electrical_series_sampling_frequency`?